### PR TITLE
Support more device type in builder

### PR DESCRIPTION
**Pull request by futuretea**
Signed-off-by: futuretea <Hang.Yu@suse.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Support more device type in builder
- input
- hostDevices
- GPU
- TPM

will use these methods in terraform-provider-harvester

**Related Issue:**
https://github.com/harvester/harvester/issues/3119
https://github.com/harvester/harvester/issues/2910

**Test plan:**

### CODE REVIEW

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

File: pkg/builder/device.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	InputBusUSB     = "usb"
+	InputBusVirtio  = "virtio"
+)
+
+func (v *VMBuilder) Input(inputName, inputType, inputBus string) *VMBuilder {
+	inputs := v.VirtualMachine.Spec.Template.Spec.Domain.Devices.Inputs
+	input := kubevirtv1.Input{
+		Name: inputName,
+		Type: inputType,
+		Bus:  inputBus,
```
Comment (guangbochen) [inline pkg/builder/device.go]: Bus field is optional, does empty string works?
Also, should we add validations here of other fields?

File: pkg/builder/device.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	v.VirtualMachine.Spec.Template.Spec.Domain.Devices.Inputs = inputs
+	return v
+}
+
+func (v *VMBuilder) HostDevice(name, hostDeviceName, tag string) *VMBuilder {
+	hostDevices := v.VirtualMachine.Spec.Template.Spec.Domain.Devices.HostDevices
+	hostDevice := kubevirtv1.HostDevice{
+		Name:       name,
+		DeviceName: hostDeviceName,
+		Tag:        tag,
```
Comment (guangbochen) [inline pkg/builder/device.go]: ditto

File: pkg/builder/device.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	return v
+}
+
+func (v *VMBuilder) GPU(name, hostDeviceName, tag string, virtualGPUOptions *kubevirtv1.VGPUOptions) *VMBuilder {
+	gpus := v.VirtualMachine.Spec.Template.Spec.Domain.Devices.GPUs
+	gpu := kubevirtv1.GPU{
+		Name:              name,
+		DeviceName:        hostDeviceName,
+		Tag:               tag,
+		VirtualGPUOptions: virtualGPUOptions,
```
Comment (guangbochen) [inline pkg/builder/device.go]: ditto

File: pkg/builder/device.go (hunk 1 above)
Comment (futuretea) [reply pkg/builder/device.go]: Updated. 
Prefer to add validations in terraform-provider-harvester

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**